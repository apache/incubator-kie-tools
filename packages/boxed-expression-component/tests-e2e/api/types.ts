import {
  DMN15__tConditional,
  DMN15__tContext,
  DMN15__tDecisionTable,
  DMN15__tFilter,
  DMN15__tFor,
  DMN15__tFunctionDefinition,
  DMN15__tInvocation,
  DMN15__tList,
  DMN15__tLiteralExpression,
  DMN15__tQuantified,
  DMN15__tRelation,
} from "@kie-tools/dmn-marshaller/dist/schemas/dmn-1_5/ts-gen/types";

export type BoxedLiteral = { __$$element: "literalExpression" } & DMN15__tLiteralExpression;
export type BoxedRelation = { __$$element: "relation" } & DMN15__tRelation;
export type BoxedContext = { __$$element: "context" } & DMN15__tContext;
export type BoxedDecisionTable = { __$$element: "decisionTable" } & DMN15__tDecisionTable;
export type BoxedList = { __$$element: "list" } & DMN15__tList;
export type BoxedInvocation = { __$$element: "invocation" } & DMN15__tInvocation;
export type BoxedFunction = { __$$element: "functionDefinition" } & DMN15__tFunctionDefinition;
export type BoxedFor = { __$$element: "for" } & DMN15__tFor;
export type BoxedEvery = { __$$element: "every" } & DMN15__tQuantified;
export type BoxedSome = { __$$element: "some" } & DMN15__tQuantified;
export type BoxedConditional = { __$$element: "conditional" } & DMN15__tConditional;
export type BoxedFilter = { __$$element: "filter" } & DMN15__tFilter;

export type Normalized<T> = WithRequiredDeep<T, "@_id">;

type WithRequiredDeep<T, K extends keyof any> = T extends undefined
  ? T
  : T extends Array<infer U>
    ? Array<WithRequiredDeep<U, K>>
    : { [P in keyof T]: WithRequiredDeep<T[P], K> } & (K extends keyof T
        ? { [P in K]-?: NonNullable<WithRequiredDeep<T[P], K>> }
        : T);
