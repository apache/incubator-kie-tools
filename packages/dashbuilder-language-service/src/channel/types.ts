/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { TextDocument } from "vscode-languageserver-textdocument";
import { CompletionItemKind, Range } from "vscode-languageserver-types";

export declare type DashbuilderLsNodeType = "object" | "array" | "property" | "string" | "number" | "boolean" | "null";
export declare type DashbuilderJsonPath = (string | number)[];

/**
 * The AST node used in the LanguageServices
 */
export type DashbuilderLsNode = {
  type: DashbuilderLsNodeType;
  value?: any;
  offset: number;
  length: number;
  colonOffset?: number;
  parent?: DashbuilderLsNode;
  children?: DashbuilderLsNode[];
};

export interface TranslateArgs {
  completion: object | string;
  completionItemKind: CompletionItemKind;
  currentNodeRange?: Range;
  cursorOffset: number;
  document: TextDocument;
  overwriteRange?: Range;
}
