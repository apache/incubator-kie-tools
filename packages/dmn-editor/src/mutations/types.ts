import { NodeType } from "../diagram/connections/graphStructure";
import { NODE_TYPES } from "../diagram/nodes/NodeTypes";

export enum NodeNature {
  DRG_ELEMENT = "DRG_ELEMENT",
  ARTIFACT = "ARTIFACT",
}

export const nodeNatures: Record<NodeType, NodeNature> = {
  [NODE_TYPES.inputData]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.decision]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.bkm]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.knowledgeSource]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.decisionService]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.textAnnotation]: NodeNature.ARTIFACT,
  [NODE_TYPES.group]: NodeNature.ARTIFACT,
};
