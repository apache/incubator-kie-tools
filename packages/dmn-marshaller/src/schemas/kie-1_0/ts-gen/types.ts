// This file was automatically generated

import { XmlParserTsRootElementBaseType } from "@kie-tools/xml-parser-ts";

export type KIE__tComponentsWidthsExtension = {
  ComponentWidths?: KIE__tComponentWidths[]; // from type KIE__tComponentsWidthsExtension @ KIE.xsd
};

export type KIE__tComponentWidths = {
  "@_dmnElementRef"?: string; // from type KIE__tComponentWidths @ KIE.xsd
  width?: number[]; // from type KIE__tComponentWidths @ KIE.xsd
};

export type KIE__tAttachment = {
  "@_url"?: string; // from type KIE__tAttachment @ KIE.xsd
  "@_name"?: string; // from type KIE__tAttachment @ KIE.xsd
};
