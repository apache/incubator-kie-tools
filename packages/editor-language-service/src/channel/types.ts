/*
 * Copyright 2023 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Position, TextDocument } from "vscode-languageserver-textdocument";
import { CompletionItemKind, Range } from "vscode-languageserver-types";
import { EditorLanguageServiceCommandTypes } from "../api";

// types ElsJsonPath, ELsNode, ELsNodeType need to be compatible with jsonc types
export declare type ElsJsonPath = (string | number)[];
export declare type ELsNodeType = "object" | "array" | "property" | "string" | "number" | "boolean" | "null";

type OmitDistributive<T, K extends PropertyKey> = T extends any
  ? T extends object
    ? Id<OmitRecursively<T, K>>
    : T
  : never;
type Id<T> = {} & { [P in keyof T]: T[P] };
export type OmitRecursively<T extends any, K extends PropertyKey> = Omit<
  { [P in keyof T]: OmitDistributive<T[P], K> },
  K
>;

/**
 * The AST node used in the LanguageServices
 */
export type ELsNode = {
  type: ELsNodeType;
  value?: any;
  offset: number;
  length: number;
  colonOffset?: number;
  parent?: ELsNode;
  children?: ELsNode[];
};

export interface ShouldCompleteArgs {
  content: string;
  cursorOffset: number;
  cursorPosition: Position;
  node: ELsNode | undefined;
  path: ElsJsonPath;
  root: ELsNode | undefined;
}

export interface ShouldCreateCodelensArgs {
  content: string;
  node: ELsNode;
  commandName: EditorLanguageServiceCommandTypes;
}

export interface TranslateArgs {
  completion: object | string;
  completionItemKind: CompletionItemKind;
  currentNodeRange?: Range;
  cursorOffset: number;
  document: TextDocument;
  overwriteRange?: Range;
}

export interface CodeCompletionStrategy {
  translate(args: TranslateArgs): string;
  formatLabel(label: string, completionItemKind: CompletionItemKind): string;
  shouldComplete(args: ShouldCompleteArgs): boolean;
  getStartNodeValuePosition(document: TextDocument, node: ELsNode): Position | undefined;
  shouldCreateCodelens(args: ShouldCreateCodelensArgs): boolean;
}
