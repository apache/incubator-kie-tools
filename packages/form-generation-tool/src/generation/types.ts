/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export interface FormSchema {
  name: string;
  schema: any;
}

export enum FormStyle {
  PATTERNFLY = "patternfly",
  BOOTSTRAP = "bootstrap",
}

export enum FormAssetType {
  HTML = "html",
  TSX = "tsx",
}

export interface FormAsset {
  id: string;
  assetName: string;
  content: string;
  type: FormAssetType | string;
}

export interface FormGenerationTool {
  type: string;
  generate: (schema: FormSchema) => FormAsset;
}
