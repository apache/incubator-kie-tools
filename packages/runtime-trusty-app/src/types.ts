/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
export interface TrustyContextValue {
  config: {
    counterfactualEnabled: boolean;
    explanationEnabled: boolean;
    serverRoot: string;
    basePath: string;
    useHrefLinks: boolean;
  };
}

export type RemoteData<E, D> =
  | { status: RemoteDataStatus.NOT_ASKED }
  | { status: RemoteDataStatus.LOADING }
  | { status: RemoteDataStatus.FAILURE; error: E }
  | { status: RemoteDataStatus.SUCCESS; data: D };

export enum RemoteDataStatus {
  NOT_ASKED,
  LOADING,
  FAILURE,
  SUCCESS,
}

export interface Execution {
  executionId: string;
  executionDate: string;
  executedModelName: string;
  executionType: string;
  executionSucceeded: boolean;
  executorName: string;
}

export interface Executions {
  headers: Execution[];
  total: number;
}

export interface ExecutionRouteParams {
  executionId: string;
  executionType: string;
}

export interface ItemObject {
  name: string;
  value: ItemObjectValue;
}

export type ItemObjectValue = ItemObjectUnit | ItemObjectCollection | ItemObjectStructure;

export interface ItemObjectUnit {
  kind: "UNIT";
  type: string;
  value: string | number | boolean | Array<string | number | boolean> | null;
}

export interface ItemObjectCollection {
  kind: "COLLECTION";
  type: string;
  value: ItemObjectValue[] | null;
}

export type ItemObjectMap = { [key: string]: ItemObjectValue };

export interface ItemObjectStructure {
  kind: "STRUCTURE";
  type: string;
  value: ItemObjectMap;
}

export interface InputRow {
  inputLabel: string;
  inputValue?: ItemObject["value"];
  key: string;
  category: string;
}

export enum evaluationStatus {
  EVALUATING = "Evaluating",
  FAILED = "Failed",
  NOT_EVALUATED = "Not evaluated",
  SKIPPED = "Skipped",
  SUCCEEDED = "Succeeded",
}

export type evaluationStatusStrings = keyof typeof evaluationStatus;

export interface Outcome {
  outcomeId: string;
  outcomeName: string;
  outcomeResult: ItemObjectValue;
  evaluationStatus: evaluationStatusStrings;
  hasErrors: boolean;
  messages: string[];
}

export interface FeatureScores {
  featureName: string;
  featureScore: number;
}

export interface Saliency {
  outcomeId: string;
  featureImportance: FeatureScores[];
}

export enum SaliencyStatus {
  SUCCEEDED = "SUCCEEDED",
  FAILED = "FAILED",
}

export type SaliencyStatusStrings = keyof typeof SaliencyStatus;

export interface Saliencies {
  status: SaliencyStatusStrings;
  statusDetail: string;
  saliencies: Saliency[];
}

export interface ServiceIdentifier {
  groupId?: string;
  artifactId?: string;
  version?: string;
}

export interface ModelData {
  deploymentDate?: string;
  modelId?: string;
  name: string;
  namespace: string;
  modelVersion: string;
  dmnVersion: string;
  serviceIdentifier: ServiceIdentifier;
  model: string;
}

export interface CFSearchInput {
  name: string;
  value: CFSearchInputValue;
}

export type CFSearchInputValue = CFSearchInputUnit | CFSearchInputCollection | CFSearchInputStructure;

export interface CFSearchInputUnit {
  kind: "UNIT";
  type: string;
  fixed?: boolean;
  domain?: CFNumericalDomain | CFCategoricalDomain;
  originalValue: ItemObject["value"];
}

export interface CFNumericalDomain {
  type: "RANGE";
  lowerBound?: number;
  upperBound?: number;
}

export interface CFCategoricalDomain {
  type: "CATEGORICAL";
  categories: string[];
}

export interface CFSearchInputCollection {
  kind: "COLLECTION";
  type: string;
  value: Array<CFSearchInputValue> | null;
}

export type CFSearchInputValueMap = { [key: string]: CFSearchInputValue };

export interface CFSearchInputStructure {
  kind: "STRUCTURE";
  type: string;
  value: CFSearchInputValueMap | null;
}

export enum CFGoalRole {
  UNSUPPORTED,
  ORIGINAL,
  FIXED,
  FLOATING,
}

export interface CFGoal {
  id: string;
  name: string;
  role: CFGoalRole;
  value: ItemObjectValue;
  originalValue: CFGoal["value"];
}

export type CFResult = Array<unknown>;

export interface CFStatus {
  isDisabled: boolean;
  executionStatus: CFExecutionStatus;
  lastExecutionTime: null | string;
}

export enum CFExecutionStatus {
  COMPLETED,
  RUNNING,
  NOT_STARTED,
  FAILED,
  NO_RESULTS,
}

export type CFAnalysisResetType = "NEW" | "EDIT";

export interface CFAnalysisExecution {
  executionId: string;
  counterfactualId: string;
  maxRunningTimeSeconds: number;
}

export interface CFAnalysisResult {
  executionId: string;
  counterfactualId: string;
  type: "counterfactual";
  valid: boolean;
  status: "SUCCEEDED" | "FAILED";
  statusDetails: string;
  solutionId: string;
  isValid: boolean;
  stage: "INTERMEDIATE" | "FINAL";
  inputs: ItemObject[];
  outputs: ItemObject[];
  sequenceId: number;
}

export interface CFAnalysisResultsSets extends CFAnalysisExecution {
  goals: CFGoal[];
  searchDomains: CFSearchInput[];
  solutions: CFAnalysisResult[];
}

export type CFSupportMessage = {
  id: string;
  message: string;
};
