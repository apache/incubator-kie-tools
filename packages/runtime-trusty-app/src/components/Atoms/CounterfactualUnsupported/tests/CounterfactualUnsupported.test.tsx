/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from "react";
import { shallow } from "enzyme";
import CounterfactualUnsupported from "../CounterfactualUnsupported";
import { CFSupportMessage } from "../../../../types";

describe("CounterfactualUnsupported status", () => {
  test("renders messages", () => {
    const messages: CFSupportMessage[] = [
      { id: "msg1", message: "message1" },
      { id: "msg2", message: "message2" },
    ];

    const wrapper = shallow(<CounterfactualUnsupported messages={messages} />);

    expect(wrapper).toMatchSnapshot();

    const items = wrapper.find("EmptyStateBody p");
    expect(items.length).toBe(2);
    expect(items.get(0).props["data-ouia-component-id"]).toBe("msg1");
    expect(items.get(1).props["data-ouia-component-id"]).toBe("msg2");
  });
});
