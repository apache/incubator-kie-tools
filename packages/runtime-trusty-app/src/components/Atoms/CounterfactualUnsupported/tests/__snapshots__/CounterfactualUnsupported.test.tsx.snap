// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterfactualUnsupported status renders messages 1`] = `
<EmptyState
  variant="full"
>
  <EmptyStateIcon
    icon={[Function]}
  />
  <Title
    headingLevel="h1"
  >
    Counterfactuals cannot be generated
  </Title>
  <EmptyStateBody>
    <p
      data-ouia-component-id="msg1"
      key="msg1"
    >
      message1
    </p>
    <p
      data-ouia-component-id="msg2"
      key="msg2"
    >
      message2
    </p>
  </EmptyStateBody>
</EmptyState>
`;
